test/api/route/messages_test.dart

```diff
@@ -436,4 +453,195 @@ void main() {
       });
     });
   });
+
+  group('updateMessageFlags', () {
+    Future<UpdateMessageFlagsResult> checkUpdateMessageFlags(
+      FakeApiConnection connection, {
+      required List<int> messages,
+      required UpdateMessageFlagsOp op,
+      required MessageFlag flag,
+      required Map<String, String> expected,
+    }) async {
+      connection.prepare(json:
+        UpdateMessageFlagsResult(messages: messages).toJson());
+      final result = await updateMessageFlags(connection,
+        messages: messages, op: op, flag: flag);
+      check(connection.lastRequest).isNotNull().isA<http.Request>()
+        ..method.equals('POST')
+        ..url.path.equals('/api/v1/messages/flags')
+        ..bodyFields.deepEquals(expected);
+      return result;
+    }
+
+    test('smoke', () {
+      return FakeApiConnection.with_((connection) async {
+        await checkUpdateMessageFlags(connection,
+          messages: [1, 2, 3],
+          op: UpdateMessageFlagsOp.add, flag: MessageFlag.read,
+          expected: {
+            'messages': jsonEncode([1, 2, 3]),
+            'op': 'add',
+            'flag': 'read',
+          });
+      });
+    });
+  });
+
+  group('updateMessageFlagsForNarrow', () {
+    Future<UpdateMessageFlagsForNarrowResult> checkUpdateMessageFlagsForNarrow(
+      FakeApiConnection connection, {
+      required Anchor anchor,
+      required int numBefore,
+      required int numAfter,
+      required ApiNarrow narrow,
+      required UpdateMessageFlagsOp op,
+      required MessageFlag flag,
+      required Map<String, String> expected,
+    }) async {
+      final result = await updateMessageFlagsForNarrow(connection,
+        anchor: anchor, numBefore: numBefore, numAfter: numAfter,
+        narrow: narrow, op: op, flag: flag);
+      check(connection.lastRequest).isNotNull().isA<http.Request>()
+        ..method.equals('POST')
+        ..url.path.equals('/api/v1/messages/flags/narrow')
+        ..bodyFields.deepEquals(expected);
+      return result;
+    }
+    final fakeResult = UpdateMessageFlagsForNarrowResult(
+      processedCount: 11, updatedCount: 11, foundOldest: true,
+      foundNewest: true);
```

Comment:
nit: foundOldest and foundNewest are closely related, so go together:
```suggestion
      processedCount: 11, updatedCount: 11,
      foundOldest: true, foundNewest: true);
```